doc/adrs/194_improve_diagram_modals.adoc

```diff
@@ -0,0 +1,189 @@
+
+It will be useful if the left or right panel are expanded/minimized to always have the correct bounds.
+
+It could be also useful if we want to be more precise about the allowed bounds in the future, we could for example exclude the diagram panel from the available bounds.
+
+We will add a custom hook that will be used in a modal like this
+
+[source,typescript]
+----
+  const { getUpdatedModalPosition } = useGetUpdatedModalPosition();
+  const position = getUpdatedModalPosition(initialPosition, nodeRef);
+
+	  return (
+    <Draggable nodeRef={nodeRef} disabled={true} position={position}>
+      <Paper ref={nodeRef} className={classes.palette}>
+        <ClickAwayListener mouseEvent="onPointerDown" onClickAway={closeDialog}>
+					<MODALCONTENT></MODALCONTENT>
+        </ClickAwayListener>
+      </Paper>
+    </Draggable>
+  );
+----
+
+[source,typescript]
+----
+import { ReactFlowState, useStore, XYPosition } from '@xyflow/react';
+import { UseGetUpdatedModalPositionValue } from './useGetUpdatedModalPosition.types';
+const xyFlowDomBoundsSelector = (state: ReactFlowState): DOMRect | undefined => state.domNode?.getBoundingClientRect();
+
+export const useGetUpdatedModalPosition = (): UseGetUpdatedModalPositionValue => {
+  const xyFlowDomBounds = useStore(xyFlowDomBoundsSelector);
+  const getUpdatedModalPosition = (initialPosition: XYPosition, modalRef: React.RefObject<HTMLDivElement>) => {
+    let position: XYPosition = { ...initialPosition };
+    if (modalRef && modalRef.current && xyFlowDomBounds) {
+      const modalBounds: DOMRect = modalRef.current.getBoundingClientRect();
+      if (modalBounds) {
+        if (initialPosition.x + modalBounds.width > xyFlowDomBounds.width) {
+          position.x = xyFlowDomBounds.width - modalBounds.width;
+        }
+
+        if (initialPosition.y + modalBounds.height > xyFlowDomBounds.height) {
+          position.y = xyFlowDomBounds.height - modalBounds.height;
+        }
+      }
+    }
+    return position;
+  };
+
+  return { getUpdatedModalPosition };
+};
+----
+
+
+=== Modal UI
+
+All modals should have this shape and a component `ModalHeader` should be added.
+
+[source,typescript]
+----
+  return (
+    <Draggable nodeRef={nodeRef} disabled={true} position={position}>
+      <Paper ref={nodeRef} className={classes.palette}>
+        <ClickAwayListener mouseEvent="onPointerDown" onClickAway={closeDialog}>
+          <Stack>
+          <ModalHeader closeDialog={closeDialog} headerTitle={headerTitle}></ModalHeader>
+          <MODALCONTENT></MODALCONTENT>
+          </Stack>
+        </ClickAwayListener>
+      </Paper>
+    </Draggable>
+  );
+----
+
+Since `Draggable` and `ClickAwayListener` are used to manage the dragging events and the close events, we could have a component `ModalWrapper` managing just that.
+
+[source,typescript]
+----
+  return (
+    <ModalWrapper initialPosition={position} height={200} width={150} closeDialog={closeDialog} isDraggable={true}>
+      <Stack>
```

Comment:
Very minor comment but I'm really not fan of relying on basic MUI components like stack and grid instead of raw CSS (flexbox / grid) for the layout.